### DISCUSSION TITLE
FastAPI is getting confused between /users/{userId} and users/me endpoint

### DISCUSSION THREAD

**Discussion by knakra92** [fastapi/fastapi#9179] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import APIRouter, HTTPException, Body, Path, Cookie

router = APIRouter(
    prefix="/users",
    tags=["Users"]
)

@router.get(path="/{user_id}", status_code=200)
def get_user_by_id(user_id: str = Path(title="User Identifier")):
    print("get user by id called")
    user = users.get(user_id)

    if user:
        return user
    else:
        raise HTTPException(status_code=404, detail={"error": "User not found"})

@router.get(path="/me")
def get_data():
    return {"userInfo": "my_user"}
```

### Description

I have 2 endpoints in my code:
* /users/{userId}
* /users/me

When I'm trying to call the /users/me endpoint it is treating "me" as the userId and always calling /users/{userId} endpoint but it starts working when I change the /users/me endpoint to /users/me/ and call /users/me/ endpoint or when I move the /users/me endpoint before /users/{userId} endpoint

Can someone please confirm if that is an expected behavior?

### Operating System

macOS

### Operating System Details

<img width="472" alt="image" src="https://user-images.githubusercontent.com/32999806/222874730-5daf7105-4ef7-46c7-9b5f-eaab7cbd4a0b.png">

### FastAPI Version

0.92.0

### Python Version

3.11

**Comment by abyesilyurt** [comment, marked as answer]
This is an expected behavior and the order matters! You need to define `/users/me` before `/users/{userId}`.

See the documentation for the details: https://fastapi.tiangolo.com/tutorial/path-params/#order-matters

**Reply by knakra92** [reply]
> This is an expected behaviur and the order matters! You need to define `/users/me` before `/users/{userId}`.
> 
> See the documentation for the details: https://fastapi.tiangolo.com/tutorial/path-params/#order-matters

Thank you @abyesilyurt! Appreciate it!
